lib/logstash/outputs/rabbitmq.rb

```diff
@@ -1,188 +1,59 @@
-    @hare_info.exchange.publish(message, :routing_key => event.sprintf(@key), :properties => { :persistent => @persistent })
-  rescue MarchHare::Exception, IOError, com.rabbitmq.client.AlreadyClosedException => e
-    return if terminating?
-
-    @logger.error("Error while publishing. Will retry.",
-                  :message => e.message,
-                  :exception => e.class,
-                  :backtrace => e.backtrace)
-
-    sleep_for_retry
-    retry
-  end
-
-  public
-  def to_s
-    return "<LogStash::RabbitMQ::Output: amqp://#{@user}@#{@host}:#{@port}#{@vhost}/#{@exchange_type}/#{@exchange}\##{@key}>"
-  end
-
-  public
-  def teardown
-    @hare_info.connection.close if connection_open?
-
-    finished
-  end
-
-  private
-  def settings
-    return @settings if @settings
-
-    s = {
-      :vhost => @vhost,
-      :host  => @host,
-      :port  => @port,
-      :user  => @user,
-      :automatic_recovery => @automatic_recovery,
-      :pass => @password ? @password.value : "guest",
-    }
-    s[:tls] = @ssl if @ssl
-    @settings = s
-  end
-
-  private
-  def connect
-    @logger.debug("Connecting to RabbitMQ. Settings: #{settings.inspect}, queue: #{@queue.inspect}")
-
-
-    connection = MarchHare.connect(settings)
-    connection.on_blocked { @logger.warn("RabbitMQ output blocked! Check your RabbitMQ instance!") }
-    connection.on_unblocked { @logger.warn("RabbitMQ output unblocked!") }
-
-    channel = connection.create_channel
-    @logger.info("Connected to RabbitMQ at #{settings[:host]}")
-
-    exchange = declare_exchange!(channel)
-
-    HareInfo.new(connection, channel, exchange)
-  end
-
-  private
-  def declare_exchange!(channel)
-    @logger.debug("Declaring an exchange", :name => @exchange,
-                  :type => @exchange_type, :durable => @durable)
-    exchange = channel.exchange(@exchange, :type => @exchange_type.to_sym, :durable => @durable)
-    @logger.debug("Exchange declared")
-    exchange
-  rescue StandardError => e
-    @logger.error("Could not declare exchange!",
-                  :exchange => @exchange, :type => @exchange_type,
-                  :durable => @durable, :error_class => e.class.name,
-                  :error_message => e.message, :backtrace => e.backtrace)
-    raise e
-  end
-
-  private
-  def connect!
-    @hare_info = connect()
-  rescue MarchHare::Exception => e
-    return if terminating?
-
-    @logger.error("RabbitMQ connection error, will retry.",
-                  :message => e.message,
-                  :exception => e.class.name,
-                  :backtrace => e.backtrace)
-
-    sleep_for_retry
-    retry
-  end
-
-  private
-  def connection_open?
-    @hare_info && @hare_info.connection && @hare_info.connection.open?
-  end
-
-  private
-  def sleep_for_retry
-    sleep @connect_retry_interval
+require "logstash/devutils/rspec/spec_helper"
+require "logstash/pipeline"
+require "logstash/plugin_mixins/rabbitmq_connection"
+
+module LogStash
```

Comment:
I know it doesn't say much but can you add the block of documentation before the module? The doc tool will pick it up.

```
# Push events to a RabbitMQ exchange. Requires RabbitMQ 2.x
# or later version (3.x is recommended).
#
# Relevant links:
#
# * http://www.rabbitmq.com/[RabbitMQ]
# * http://rubymarchhare.info[March Hare]